server/catgenome/src/main/java/com/epam/catgenome/manager/project/ProjectManager.java

```diff
@@ -388,7 +394,38 @@ public Project create(final Project project, Long parentId) {
                 dataItems.forEach(i -> fillFileTypeLists(i, newGeneFiles, newVcfFiles));
             }
         }
-
+        if (!newProject && !CollectionUtils.isEmpty(loadedProject.getNotes())) {
+            List<Long> existingNoteIds = loadedProject.getNotes().stream()
+                    .map(ProjectNote::getNoteId)
+                    .collect(Collectors.toList());
+            List<Long> newNoteIds = helpProject.getNotes() == null ? Collections.emptyList() :
+                    helpProject.getNotes().stream()
+                    .map(ProjectNote::getNoteId)
+                    .collect(Collectors.toList());
+            List<Long> notesToDelete = existingNoteIds.stream()
+                    .filter(o -> !newNoteIds.contains(o))
+                    .collect(Collectors.toList());
+            if (!notesToDelete.isEmpty()) {
+                final Filter filter = Filter.builder()
```

Comment:
If `noteId` is not unique for all projects, we need to add `projectId` filter as well